feat(textarea): initial design guidance (#34)
vmware-clarity/clarity-guidance: 1033-textarea-design-guidance.md

---
category: component design guidance
state: draft
created: 2024-10-18
updated: 2024-10-22
---

# Textarea Design Guidance

The [textarea](https://clarity.design/documentation/textarea) component enables the user to input multiple lines of text.

## Guidance

> - **Must** use a textarea when the expected user input may be more than a single line. If only a single line is expected, use an [input](https://clarity.design/documentation/input) instead.

Reviewer: ```suggestion
- Design teams **must** use a textarea when the expected user input spans multiple lines. If only a single line is needed, use an [input](https://clarity.design/documentation/input) instead. This ensures the field aligns with user expectations and encourages appropriate input behavior.
```

> - **Should** provide a clear and concise visible label for context.

Reviewer: ```suggestion
- Design teams **should** provide a clear and concise visible label for context, ensuring users understand what input is expected and reducing the chance of errors.
```

> - **Should not** be a substitute for a proper label element. It **may** be used to show an example of the type of data that should be entered into a form, but it **should not** be used in place of helper text.

Reviewer: ```suggestion
- Design teams **must not** use a textarea as a substitute for a proper label element. Labels are essential for accessibility and ensure users clearly understand the purpose of the field. A textarea **may** be used to show an example of the expected input but should not replace helper text intended to guide the user.
```

---
category: component design guidance
state: draft
created: 2024-10-18
updated: 2024-10-21
---

# Textarea Design Guidance

> [Textarea](https://clarity.design/documentation/textarea) component enable the user to input a single sentence or less of text.

Reviewer 2: ```suggestion
The [textarea](https://clarity.design/documentation/textarea) component enables the user to input multiple lines of text.
```

## Guidance

- **Must** use Textarea when the expected user input may be more than a single line. If only a single line is expected, use [Input](https://clarity.design/documentation/input) instead.
- **Should** provide a clear and concise label for context.

> - Placeholders **should** NOT be a substitute for a proper label element. It may only be used to show an example of the type of data that should be entered into a form. It **should** NOT be used in place of helper text either.

Reviewer 2: ```suggestion
- Placeholders **should not** be a substitute for a proper label element. It may only be used to show an example of the type of data that should be entered into a form. It **should not** be used in place of helper text either.
```

> - Placeholders **should not** be a substitute for a proper label element. It may only be used to show an example of the type of data that should be entered into a form. It **should not** be used in place of helper text either.

Reviewer 2: ```suggestion
- Placeholders **should not** be a substitute for a proper label element. It **may** only be used to show an example of the type of data that should be entered into a form. It **should not** be used in place of helper text either.
```

"May" is guidance keyword. It should be bold. Example: https://google.aip.dev/8

Reviewer 2: ```suggestion
- Placeholders **should not** be a substitute for a proper label element. It **may** be used to show an example of the type of data that should be entered into a form, but it **should not** be used in place of helper text.
```

Actually, "may only" should be "must" instead since it is not optional. If this is truly a "may" thing, drop the word "only". I also suggest some other rewording to drop the word "either" at the end.

- [States supported for Textarea](https://clarity.design/documentation/textarea#states).

> - Textarea is available in vertical, horizontal and compact layouts. [Layouts supported for Textarea](https://clarity.design/documentation/textarea#layouts)

Reviewer 2: These two lines are not guidance. This is just documentation of available features. If we don't have guidance for when to use a specific state or layout, it shouldn't be here.

Author: Removed
